File: src/main/study/spark/scalaVersion/sql/DataFrameCreate.scala
package spark.scalaVersion.sql

import org.apache.spark.{SparkConf, SparkContext, sql}

object DataFrameCreate {
  def main(args: Array[String]){
    val conf = new SparkConf()
      .setAppName("WordCount")
    val sc = new SparkContext(conf)
    val sqlContext  = new sql.SQLContext(sc)
    val ds = sqlContext.read.json("hdfs://xy:8021/students.json")
    ds.show()
  }

}

File: src/main/study/spark/javaVersion/sql/DataFrameCreate.java
package spark.javaVersion.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.SQLContext;


public class DataFrameCreate {
    public static void DataFrameCreate(){
        SparkConf conf = new SparkConf()
                .setAppName("DataFrameCreate");
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);
        Dataset ds = sqlContext.read().json("hdfs://xy:8021/students.json");
        ds.show();
        ds.printSchema();
        ds.select("name").show();
        ds.select(ds.col("name"),ds.col("age").plus(1)).show();
        ds.filter(ds.col("age").gt(18)).show();
        ds.groupBy(ds.col("age")).count().show();
    }
}
